sphinx_exercise/directive.py

```diff
@@ -275,3 +278,39 @@ def run(self) -> List[Node]:
             return []
 
         return [node]
+
+
+# Gated Directives
+
+
+class SolutionStartDirective(SolutionDirective):
+    """
+    A gated directive for solution
+
+    .. solution-start:: <exercise-reference>
+       :label:
+       :class:
+       :hidden:
+    """
+
+    name = "solution-start"
+    solution_node = solution_start_node
+
+
+class SolutionEndDirective(SolutionDirective):
+    """
+    A gated directive for solution
+
+    .. solution-end:: <exercise-reference>
+       :label:
```

Comment:
Why would anybody use `:label:` `:class:` etc here? I feel like we shouldn't document this pattern and should probably raise an error if somebody puts config here - it seems simpler to have all configuration in the `start` part of the directive

EDIT: ah I see you plan to do this :-)

Reply:
Indeed. Need to change `solution-end` to a much simpler class. The only complicated factor is that it can contain content so we need some state parsing etc.